subsys/mgmt/mcumgr/lib/cmd/shell_mgmt/Kconfig

```diff
@@ -22,3 +22,16 @@ config MCUMGR_CMD_SHELL_MGMT
 	  on a stack, by the mcumgr, so enabling MCUMGR_CMD_SHELL_MGMT and
 	  changes to the CONFIG_SHELL_CMD_BUFF_SIZE may increase stack size
 	  requirements.
+
+if MCUMGR_CMD_SHELL_MGMT
+
+config MCUMGR_CMD_SHELL_MGMT_LEGACY_RC_RETURN_CODE
+	bool "Use legacy rc field for shell function return code"
+	help
+	  With this option, the legacy combined rc field will be used for
```

Comment:
"Enabling this options brings back legacy behaviour where shell return code is returned, incorrectly, in rc filed that has originally been designated for returning SMP processing errors.
When disabled there will be additional ret field used for shell command exit code, and rc will be used for SMP processing error codes."

Reply:
Fixed